visualization_scripts/run.py

```diff
@@ -0,0 +1,175 @@
+""" 
+script loads style file and TPS output file into Cytoscape session
+
+@author: Adam Shedivy
+
+"""
+
+# imports 
+from py2cytoscape.data.cynetwork import CyNetwork
+from py2cytoscape.data.cyrest_client import CyRestClient
+from py2cytoscape.data.style import StyleUtil
+import py2cytoscape.util.cytoscapejs as cyjs
+import py2cytoscape.cytoscapejs as renderer
+from py2cytoscape import cyrest
+import networkx as nx
+import pandas as pd
+import json
+import sys
+import time
+import subprocess
+import os
+from requests.exceptions import ConnectionError as CE
+from json.decoder import JSONDecodeError 
+
+#assert Cytoscape is running on machine
+
+def process_exists(process_name):
+    """
+    function to assert Cytoscape is running
+
+    Args:
+        process_name (str): name of process to check for
+        example - 'Cytoscape.exe'
+
+    Returns:
+        true if process runnning, else false
+
+    """
+    bytes_name = str.encode(process_name)
+    call = 'TASKLIST', '/FI', 'imagename eq %s' % process_name
+
+    # use buildin check_output right away
+    output = subprocess.check_output(call)
+
+    # check in last line for process name
+    last_line = output.strip().split(b'\r\n')[-1]
+
+    return last_line.lower().startswith(bytes_name.lower())
+
+
+def find_path(name, path):
+    """
+    Search Algorithm to find Cytoscape application on machine
+
+    Args:
+        name (str): name of application 
+        path (str): top directory to start search from 
+
+    Returns:
+        path to first match on machine
+    """
+
+    # list to hold found matches
+    result = []
+
+    # walk directory/file tree 
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            result.append(os.path.join(root, name))
+    return result[0]
+
+
+def vis(output, style):
+    """
+    REQUIRED ARGS:
+        output: output file from tps software
+        style: style file for cytoscpe 
+    """
+    
+    #create cyrest client
+    cyy = cyrest.cyclient()
+    
+    # Step 1: Create py2cytoscape client
+    cy = CyRestClient()
+
+    # Reset
+    cy.session.delete()
+    print("---session reset")
+
+    # Step 2: Load network from somewhere
+    print("---loading network file")
+    tps_net = cy.network.create_from(output)
+    print("---Done")
+    time.sleep(2)
+
+
+    # Step 3: Apply layout
+    # list of styles to apply
+    # only contains the one style loaded from style file 
+    print("---apply style file")
+    style = cyy.vizmap.load_file(style)

```

Comment:
This line caused an error in my testing
```
 File "run.py", line 100, in vis
    style = cyy.vizmap.load_file(style)
  File "C:\Users\agitter\.conda\envs\py2cyto\lib\site-packages\py2cytoscape\cyrest\vizmap.py", line 83, in load_file
    response=api(url=self.__url+"/load file", PARAMS=PARAMS, method="POST", verbose=verbose)
  File "C:\Users\agitter\.conda\envs\py2cyto\lib\site-packages\py2cytoscape\cyrest\base.py", line 150, in api
    raise ValueError(res["errors"][0])
ValueError: {'status': 500, 'type': 'urn:cytoscape:ci:cyrest-core:v1:handle-json-command:errors:2', 'message': "Don't know how to read file:/C:/Program%20Files/Cytoscape_v3.7.1/style.xml", 'link': 'file:/C:/Users/agitter/CytoscapeConfiguration/3/framework-cytoscape.log'}
```

Is this because I provided `style.xml` as an input argument instead of the full absolute path?  Python can convert the relative path to an absolute path if that's what `vizmap.load_file` expects.  I believe that is the problem, because when I tried again with the full absolute path, the network and style file loaded correctly.

Reply:
I have run into this issue in the past as well. What I found is that the `vizmap.load_file(style)` looks into the Cytoscape application directory on your machine. For instance, in the download folder called Cytoscape_v3.7.1 (I assume yours is similar), I had to move a copy of the style file into this directory in order to use relative paths such as just `style.xml` when calling the script. Providing the full pathname to the load style function seems to fix the issue.